### PR TITLE
Upgrade to latest stable OS versions.

### PR DESCRIPTION
**Pull request by jsirois**
The release workflow was already upgraded to Ubuntu 22.04 in #2127.

### CODE REVIEW

**Review by kaos — APPROVED** [review]

**Comment by jsirois** [comment]
The merge pulls in changes from #2132 ([5d25bbe](https://github.com/pantsbuild/pex/pull/2131/commits/5d25bbe5e58e164b530a4068ef33a2a4e4301223) & [b5a42c1](https://github.com/pantsbuild/pex/pull/2131/commits/b5a42c1a87fe1d7fdf97fad366ed78a6f2adc540))to fix CI; so I'll wait on that to land before landing this.

**Review by kaos — APPROVED** [review]

**Review by kaos — APPROVED** [review]

**Pull request merged**
